File: source/generic/s3eApsalar.cpp
/*
Generic implementation of the s3eApsalar extension.
This file should perform any platform-indepedentent functionality
(e.g. error checking) before calling platform-dependent implementations.
*/

/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */


#include "s3eApsalar_internal.h"
s3eResult s3eApsalarInit()
{
    //Add any generic initialisation code here
    return s3eApsalarInit_platform();
}

void s3eApsalarTerminate()
{
    //Add any generic termination code here
    s3eApsalarTerminate_platform();
}

void s3eApStart(const char* apiKey, const char* apiSecret)
{
	s3eApStart_platform(apiKey, apiSecret);
}

void s3eApRestart(const char* apiKey, const char* apiSecret)
{
	s3eApRestart_platform(apiKey, apiSecret);
}

bool s3eApStarted()
{
	return s3eApStarted_platform();
}

void s3eApEnd()
{
	s3eApEnd_platform();
}

void s3eApLogEvent(const char* name)
{
	s3eApLogEvent_platform(name);
}

s3eApDict* s3eApDictCreate()
{
	return s3eApDictCreate_platform();
}

void s3eApDictAddString(s3eApDict* dict, const char* key, const char* value)
{
	s3eApDictAddString_platform(dict, key, value);
}

void s3eApDictAddInt(s3eApDict* dict, const char* key, int value)
{
	s3eApDictAddInt_platform(dict, key, value);
}

void s3eApDictAddFloat(s3eApDict* dict, const char* key, float value)
{
	s3eApDictAddFloat_platform(dict, key, value);
}

void s3eApDictAddDict(s3eApDict* dict, const char* key, s3eApDict* value)
{
	s3eApDictAddDict_platform(dict, key, value);
}

void s3eApLogEventWithArgs(const char* name, s3eApDict* dict)
{
	s3eApLogEventWithArgs_platform(name, dict);
}

void s3eApSetFBAppId(const char* appId)
{
	s3eApSetFBAppId_platform(appId);
}

File: source/android/s3eApsalar.java
/*
java implementation of the s3eApsalar extension.

Add android-specific functionality here.

These functions are called via JNI from native code.
*/
/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */
import com.ideaworks3d.marmalade.LoaderAPI;
import com.ideaworks3d.marmalade.LoaderActivity;
import com.apsalar.sdk.Apsalar;
import org.json.JSONObject;
import org.json.JSONException;
import android.util.Log;

class s3eApsalar
{
	private static final String TAG = "APSALAR";

    public void s3eApStart(String apiKey, String apiSecret)
    {
		Log.v(TAG, "Session Start");
        Apsalar.startSession(LoaderActivity.m_Activity, apiKey, apiSecret);
    }
    public void s3eApRestart(String apiKey, String apiSecret)
    {
		Log.v(TAG, "Session ReStart");
        Apsalar.restartSession(LoaderActivity.m_Activity, apiKey, apiSecret);
    }
    public boolean s3eApStarted()
    {
        return !"None".equals(Apsalar.getSessionId());
    }
    public void s3eApEnd()
    {
    }
    public void s3eApLogEvent(String name)
    {
        Apsalar.event(name);
    }
    public JSONObject s3eApDictCreate()
    {
        return new JSONObject();
    }
    public void s3eApDictAddString(JSONObject dict, String key, String value)
    {
		try {
			dict.put(key, value);
		} catch(JSONException e) {
			Log.v(TAG, "JSONException in s3eApDictAddString");
		}
    }
    public void s3eApDictAddInt(JSONObject dict, String key, int value)
    {
        try {
			dict.put(key, value);
		} catch(JSONException e) {
			Log.v(TAG, "JSONException in s3eApDictAddInt");
		}
    }
    public void s3eApDictAddFloat(JSONObject dict, String key, float value)
    {
        try {
			dict.put(key, value);
		} catch(JSONException e) {
			Log.v(TAG, "JSONException in s3eApDictAddFloat");
		}
    }
    public void s3eApDictAddDict(JSONObject dict, String key, JSONObject value)
    {
        try {
			dict.put(key, value);
		} catch(JSONException e) {
			Log.v(TAG, "JSONException in s3eApDictAddDict");
		}
    }
    public void s3eApLogEventWithArgs(String name, JSONObject dict)
    {
        Apsalar.eventJSON(name, dict);
    }
	public void s3eApSetFBAppId(String appId)
    {
        Apsalar.setFBAppId(appId);
    }
}
